### ISSUE TITLE
Investigating the new version list mechanism

### ISSUE THREAD

**Issue by stefandollase** [toolbox4minecraft/amidst#78]
As described in #77, @BigAlanM discovered that Mojang is about to change the web API to request version information.

Dinnerbone added the following comment to [the old versions.json](https://s3.amazonaws.com/Minecraft.Download/versions/versions.json):

> "__comment": "This URL is being phased out! Please update your scripts to check https://launchermeta.mojang.com/mc/game/version_manifest.json instead. Thanks <3 --Dinnerbone",

He must have added this when he adjusted the release mechanism as announced in [this tweet](https://twitter.com/Dinnerbone/status/694801682883035136).

It seems like we have to further investigate this and adjust the code that reads the version list. I am not sure whether it is enough to adjust the URL. Also, in the new version list, there is a new entry for each profile called `url`. This seems to point to the version's json file which contains information about which libraries need to be loaded for the given Minecraft version. Amidst already uses this information. We have to investigate if the format was changed. Maybe, we can also extract some more useful information.

We should also update the version list included in the Amidst jar file.

**Comment by BigAlanM** [comment]
Not being a modern  language programmer, but I suspect that this is part of Mojang's promise to make the Minecraft API more friendly to mod developers that I saw rumored at one time.

**Comment by stefandollase** [comment]
I looked into this again. It seems like the new version list file contains all information we used from the old one in exactly the same format. This means for now it should just work to exchange the URL. I will do this before the stable release.

- [ ] update version list url
- [ ] update version list in the resource files for offline usage

---

However, there are more information available which I would like to use, especially the `url`: Currently, the devtools download only the jar files, but not the libraries. Now, we get an explicit url for the jar files, the libraries and I guess for everything that is downloaded by the launcher to install a new Minecraft version. This is very useful, for mod developers, but also for us when testing stuff against all versions that are available in the launcher.

Also, I would like to change the version mechanism use by Amidst to something more flexible: The magic string is really not that helpful for identifying individual versions. We can try to find something more reliable by ourselves or we can just use the `sha1` and `size` values given in Mojang's new version list. I guess, we should go with the size and sha1 values, since the information needed for this are directly provided by Mojang and used by many more tools than Amidst. I would also like to change the version mechanism to be data driven instead of having everything hard-coded in an enum. Since this is quite a lot work to do, I will not do it before the stable release of Amidst v4.0. The advantage would be, that we would only need to release a new version of Amidst when there are changes to the algorithms. We would not need a new release for every new release of Minecraft, just to add the magic string to the enum.

This is just an outlook. There are still some things to discuss. However, I will create a new issue for this when I get the time. This issue should only be about adapting Amidst to Mojang's new version list.
